### PR TITLE
Sommelier 4

### PR DESCRIPTION
**Pull request by Maximus135**

### CODE REVIEW

**Review by Mixxx-a — APPROVED** [review]

File: App/components/Input/input.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  size,
+  value,
+  onChange,
+  clearButtonMode,
+  maxLength,
+}: Props) => {
+  const [text, setText] = useState(value)
+  return (
+    <Styled.StyledView editable={editable} size={size}>
+      <SvgUri
```
Comment (StanislavMyakishev) [inline App/components/Input/input.tsx]: Change usage of SvgUri onto the new package one

File: App/components/Input/input.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        defaultValue={defaultValue}
+        placeholder={placeholder}
+        editable={editable}
+        size={size}
+        value={text}
+        onChange={onChange}
+        onChangeText={(text) => setText(text)}
+        maxLength={maxLength}
+      />
+      {clearButtonMode ? (
```
Comment (StanislavMyakishev) [inline App/components/Input/input.tsx]: maybe it's better to use { clearButtonMode && ()}

Comment (StanislavMyakishev) [reply App/components/Input/input.tsx]: [link to the docs](https://ru.reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)

File: App/components/Input/input.styled.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { TextInputProps } from 'react-native'
+import styled from 'styled-components/native'
+import theme from '../../theme/index'
+
+type Props = {
+  size: keyof typeof theme.inputSizes
+  editable: boolean
+}
+
+export const StyledInput = styled.TextInput<Props & TextInputProps>`
```
Comment (StanislavMyakishev) [inline App/components/Input/input.styled.ts]: "editable" prop is not used here

File: App/components/Input/input.styled.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  margin-left: 5px;
+  font-size: ${({ size, theme }) => theme.fontSizes[size]};
+  overflow: hidden;
+  width: 315px;
+`
+
+export const StyledView = styled.View<Props>`
+  display: flex;
+  flex-direction: row;
+  border: 1px solid #9e9e9e;
```
Comment (StanislavMyakishev) [inline App/components/Input/input.styled.ts]: please consider adding a new color to the theme

**Review by Mixxx-a — CHANGES_REQUESTED** [review]
.

File: App/components/Input/input.story.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      placeholder='Поиск по избранному'
+      editable
+    />
+  </StyledWrapper>
+)
+
+export const knobs = () => (
+  <StyledWrapper>
+    <Input
+      onChange={action('Input has been change')}
```
Comment (StanislavMyakishev) [inline App/components/Input/input.story.tsx]: *changed

File: App/components/Input/input.story.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      size='sm'
+      clearButtonMode={false}
+      placeholder='Поиск по избранному'
+      editable
+    />
+  </StyledWrapper>
+)
+
+export const knobs = () => (
+  <StyledWrapper>
```
Comment (StanislavMyakishev) [inline App/components/Input/input.story.tsx]: Please add setState() so we can use onChange func

**Review by StanislavMyakishev — CHANGES_REQUESTED** [review]

File: App/components/Input/input.tsx (hunk 2 above)
Comment (Maximus135) [reply App/components/Input/input.tsx]: done

File: App/components/Input/input.story.tsx (hunk 5 above)
Comment (Maximus135) [reply App/components/Input/input.story.tsx]: done

File: App/components/Input/input.tsx (hunk 1 above)
Comment (Maximus135) [reply App/components/Input/input.tsx]: done

File: App/components/Input/input.styled.ts (hunk 4 above)
Comment (Maximus135) [reply App/components/Input/input.styled.ts]: done

File: App/components/Input/input.styled.ts (hunk 3 above)
Comment (Maximus135) [reply App/components/Input/input.styled.ts]: done

File: App/components/Input/input.story.tsx (hunk 6 above)
Comment (Maximus135) [reply App/components/Input/input.story.tsx]: done

**Comment by Maximus135** [comment]
New pull request sent

**Pull request closed without merging**